vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthCombinedServerSelectionFragment.kt

```diff
@@ -36,67 +40,113 @@ import im.vector.app.features.onboarding.OnboardingAction
 import im.vector.app.features.onboarding.OnboardingFlow
 import im.vector.app.features.onboarding.OnboardingViewEvents
 import im.vector.app.features.onboarding.OnboardingViewState
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 import org.matrix.android.sdk.api.failure.isHomeserverUnavailable
+import reactivecircus.flowbinding.android.view.clicks
 import javax.inject.Inject
 
-class FtueAuthCombinedServerSelectionFragment @Inject constructor() : AbstractFtueAuthFragment<FragmentFtueServerSelectionCombinedBinding>() {
+class FtueAuthCombinedServerSelectionFragment @Inject constructor(
+        private val stringProvider: StringProvider,
+) : AbstractFtueAuthFragment<FragmentFtueServerSelectionCombinedBinding>() {
+
+    private lateinit var controller: Controller
 
     override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentFtueServerSelectionCombinedBinding {
-        return FragmentFtueServerSelectionCombinedBinding.inflate(inflater, container, false)
+        return FragmentFtueServerSelectionCombinedBinding.inflate(inflater, container, false).also {
+            controller = Controller(viewLifecycleOwner, it, stringProvider, errorFormatter)
```

Comment:
Can't we inject the controller? And just pass the `viewLifecycleOwner` and `views` here.

Reply:
yep this can be injected! this PR is a bit quick and dirty to show how paparazzi could be used, mainly that we would need to avoid direct view usages within fragment/activities